.claude/skills/critical-path/scripts/report.py

```
@@ -0,0 +1,822 @@
+    for task, info in task_table.items():
+        try:
+            active_slots = sum(int(kernel_id) >= 0 for kernel_id in (info.get("kernel_ids") or []))
+            logical_blocks = int(info.get("block_num"))
+        except (TypeError, ValueError) as exc:
+            raise ValueError(f"{directory / 'deps.json'}: task {task} has invalid kernel_ids/block_num") from exc
+        expected_rows = logical_blocks * active_slots
+        actual_rows = len(rows_by_task.get(task, []))
+        if actual_rows != expected_rows:
+            raise ValueError(
+                f"{records}: incomplete task {task} rows "
+                f"(joined={actual_rows}, expected={logical_blocks}×{active_slots}={expected_rows})"
```

Ruff (0.16.0)

[warning] String contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF001)